dask_remote/runner/api.py

```diff
@@ -35,52 +35,78 @@ class SchedulerInfo(BaseModel):
 
 
 def cluster_api(
-    cluster_proxy: Optional[ClusterProcessProxy] = None, fastapi_kwargs: Optional[dict] = None
+    cluster_proxy: Optional[ClusterProcessProxy] = None,
+    fastapi_kwargs: Optional[dict] = None,
+    scheduler_address: Optional[str] = None,
+    dashboard_link: Optional[str] = None,
 ) -> FastAPI:
-    """Create a FastAPI app that exposes given ClusterProcessProxy."""
+    """Create a FastAPI app that exposes given ClusterProcessProxy.
+
+    Params:
+        cluster_proxy: a `ClusterProcessProxy` for the cluster process (sic)
+        fastapi_kwargs: additional keyword arguments passed to `fastapi.FastAPI`
+        scheduler_address: override value for the RPC address used by remote clients
+        dashboard_link: override value for the HTTP link to the dashboard displayed to clients
+
+    Configuring `scheduler_address` and `dashboard_link` is necessary when the API runs
+    behind a reverse proxy, or when we want to support DNS/domain names.
+    """
     fastapi_kwargs = fastapi_kwargs or {}
     app = FastAPI(**fastapi_kwargs)
-    app.cluster = cluster_proxy
+    app.dask_cluster_proxy = cluster_proxy
+    app.dask_scheduler_address = scheduler_address
+    app.dask_dashboard_link = dashboard_link
 
-    @app.get("/", summary="Redirect to swagger docs")
+    @app.get("/", summary="Redirect to this page")
     async def root():
         """Redirect to API docs."""
         response = RedirectResponse(url=f"{app.openapi_prefix}/docs")
         return response
 
-    @app.get("/status", response_model=MessageResponse)
+    @app.get("/status", summary="Cluster status", response_model=ResponseMessage)
     async def status():
-        return MessageResponse(message=app.cluster.status)
+        """Return cluster status, e.g. 'running'."""
+        return ResponseMessage(message=app.dask_cluster_proxy.status)
 
-    @app.get("/scheduler_address", response_model=MessageResponse)
+    @app.get("/scheduler_address", summary="Scheduler address", response_model=ResponseMessage)
     async def scheduler_address():
-        return MessageResponse(message=app.cluster.scheduler_address)
+        """Return public scheduler address for use by RPC clients."""
+        scheduler_address = app.dask_scheduler_address or app.dask_cluster_proxy.scheduler_address
+        return ResponseMessage(message=scheduler_address)
 
     @app.get("/scheduler_info", response_model=SchedulerInfo)
     async def scheduler_info():
-        return SchedulerInfo(**app.cluster.scheduler_info)
+        return SchedulerInfo(**app.dask_cluster_proxy.scheduler_info)
+
+    @app.get(
+        "/dashboard_link", summary="Link to monitoring dashboard", response_model=ResponseMessage
+    )
+    async def scheduler_address():
```

Comment:
Maybe duplicated function name?

Reply:
Thanks! As aside, fastapi seems to be OK with repeated names as these are just dummy names. Especially in the case of GET/POST for same endpoint, it seems to be common to repeat the name